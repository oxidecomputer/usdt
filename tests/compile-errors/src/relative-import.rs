//! Test that we can't name types into the provider module using a relative import

// Copyright 2021 Oxide Computer Company

#![feature(asm)]
#![cfg_attr(target_os = "macos", feature(asm_sym))]

#[derive(serde::Serialize)]
struct Expected {
    x: u8
}

#[usdt::provider]
mod my_provider {
    use super::Expected;
    fn my_probe(_: Expected) {}
}

fn main() {
    my_provider::my_probe!(|| Different { x: 0 });
}
