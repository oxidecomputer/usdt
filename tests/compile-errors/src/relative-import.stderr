error: Use-statements in USDT macros cannot contain relative imports (`super`), because the generated macros may be called from anywhere in a crate. Consider using `crate` instead.
  --> src/relative-import.rs:14:9
   |
14 |     use super::Expected;
   |         ^^^^^^^^^^^^^^^

error[E0433]: failed to resolve: use of undeclared crate or module `my_provider`
  --> src/relative-import.rs:19:5
   |
19 |     my_provider::my_probe!(|| Different { x: 0 });
   |     ^^^^^^^^^^^ use of undeclared crate or module `my_provider`
